go-controller/pkg/metrics/master.go

```diff
@@ -640,45 +1021,29 @@ func (cpr *ControlPlaneRecorder) processNextItem() bool {
 	if term {
 		return false
 	}
-	cpr.processItem(i.(item))
+	cpr.processItem(i.(*item))
 	cpr.queue.Done(i)
 	return true
 }
 
-func (cpr *ControlPlaneRecorder) processItem(i item) {
+func (cpr *ControlPlaneRecorder) processItem(i *item) {
 	switch i.op {
 	case addPortBinding:
-		cpr.addPortBinding(i.old, i.t)
+		cpr.addPortBinding(i.p.old, i.p.timestamp)
 	case updatePortBinding:
-		cpr.updatePortBinding(i.old, i.new, i.t)
+		cpr.updatePortBinding(i.p.old, i.p.new, i.p.timestamp)
 	case addPod:
-		cpr.addPod(i.podUID, i.t)
+		cpr.podRecords[i.p.uid] = &record{timestamp: i.p.timestamp, timestampType: firstSeen}
 	case cleanPod:
-		cpr.cleanPod(i.podUID)
+		delete(cpr.podRecords, i.p.uid)
```

Comment:
Maybe we can also add deletion of a PodRecord after it's used the last time (which is after `metricPortBindingUpLatency.Observe`)